### DISCUSSION TITLE
prepared statement caching: are comments ignored?

### DISCUSSION THREAD

**Discussion by adriangb** [pgbouncer/pgbouncer#1044] [Q&A] [answered]
I want to used the new prepared statement functionality but I was wondering if [SQLCommenter style](https://google.github.io/sqlcommenter/spec/#introduction) comments containing trace information will break this cache. I'm guessing the answer is sadly yes, otherwise the query would have to be minimally parsed.

**Comment by JelteF** [comment]
If the comments are different, the prepared statement will be considered different. PgBouncer does not parse the query and thus does not remove the comments. So yes it sounds like that would break the cache.

**Reply by adriangb** [reply]
That makes sense. I imagine "can you please parse queries and remove comments" would be an unreasonable feature request (although in general it makes sense that the cache key would ignore comments).

Would a `query_rewrite_regex` or more targeted `prepared_statement_cache_key_regex` be a reasonable feature request?

**Reply by JelteF** [reply, marked as answer]
Why are you adding these comments to the queries in the first place? Seems like that's the problem here. Can you turn that off instead? Afaict even without pgbouncer in the middle that's going to problematic to get any perf benefit from prepared statements, because every statement will be new anyway.

**Reply by adriangb** [reply]
That’s a great point. I guess you get one or the other thinking about it from a protocol perspective. I guess we should resolve this.
